### PR TITLE
tweak metrics in txpool and node

### PR DESCRIPTION
**Pull request by libotony**
# Description

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] New and existing E2E tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have not added any vulnerable dependencies to my code

### CODE REVIEW

File: cmd/thor/node/packer_loop.go [hunk 1]
```diff
@@ -204,6 +207,10 @@ func (n *Node) pack(flow *packer.Flow) error {
 		if v, updated := n.bandwidth.Update(newBlock.Header(), time.Duration(realElapsed)); updated {
 			log.Debug("bandwidth updated", "gps", v)
 		}
+
+		metricBlockProcessedTxs().Add(int64(len(receipts)))
+		metricBlockProcessedGas().Add(int64(newBlock.Header().GasUsed()))
+		metricBlockProcessedDuration().Observe(time.Duration(realElapsed).Milliseconds())
```
Comment (otherview) [inline cmd/thor/node/packer_loop.go:213]: Does this distinguish between tx metrics from packed blocks and tx metrics from received blocks ?

File: txpool/tx_pool.go [hunk 2]
```diff
@@ -132,6 +132,7 @@ func (p *TxPool) housekeeping() {
 					p.executables.Store(executables)
 				}
 
+				metricTxPoolGauge().GaugeWithLabel(0-int64(removed), map[string]string{"source": "washed", "total": "true"})
 				log.Debug("wash done", ctx...)
```
Comment (otherview) [inline txpool/tx_pool.go:136]: Should work no ?
```suggestion
				metricTxPoolGauge().GaugeWithLabel(-int64(removed), map[string]string{"source": "washed", "total": "true"})
				log.Debug("wash done", ctx...)
```

File: bft/engine.go [hunk 3]
```diff
@@ -153,7 +153,6 @@ func (engine *BFTEngine) CommitBlock(header *block.Header, isPacking bool) error
 			return err
 		}
 		engine.casts.Mark(checkpoint, state.Quality)
-		metricBlocksCommitted().AddWithLabel(1, map[string]string{"status": "proposed"})
```
[the remark this answers is not in the document]
Comment (libotony) [reply bft/engine.go:156]: block proposed already captured in node metrics, `metricBlockProcessedCount` with label `type=proposed`

File: cmd/thor/node/packer_loop.go (hunk 1 above)
Comment (libotony) [reply cmd/thor/node/packer_loop.go:213]: No, I would consider the proposed block count as necessary but tx and gas are not.

File: bft/engine.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (otherview) [reply bft/engine.go:156]: Are processed and committed the same ?

File: cmd/thor/node/packer_loop.go (hunk 1 above)
Comment (otherview) [reply cmd/thor/node/packer_loop.go:213]: Why not? Would it not be interesting to see the tx count and gas sum of proposed blocks ?

File: bft/engine.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (libotony) [reply bft/engine.go:156]: > Are processed and committed the same ?

Not the same

[the remark this answers is not in the document]
Comment (otherview) [reply bft/engine.go:156]: I don't think this works the way I thought it worked.. That's cool.

So I guess there's no point in having `status=finalized` if it's always the same status ?

File: cmd/thor/node/packer_loop.go [hunk 4]
```diff
@@ -106,16 +106,21 @@ func (n *Node) packerLoop(ctx context.Context) {
...
+				n.txPool.Remove(tx.Hash(), tx.ID())
+			}
+			metricBlockProcessedCount().AddWithLabel(1, map[string]string{"type": "proposed", "success": "true"})
+		} else {
+			metricBlockProcessedCount().AddWithLabel(1, map[string]string{"type": "proposed", "success": "false"})
 		}
 	}()
 
-	return evalBlockProposeMetrics(n.guardBlockProcessing(flow.Number(), func(conflicts uint32) error {
+	return n.guardBlockProcessing(flow.Number(), func(conflicts uint32) error {
```
Comment (otherview) [inline cmd/thor/node/packer_loop.go:123]: I like it, it's better than what it was, however it's not a simple flow to understand.
Just a thought, but perhaps it wouldn't be a bad idea for us to consider refactoring this in the future, worth putting a todo here?

**Review by otherview — APPROVED** [review]
lgtm - before merging would consider removing the status from the `bft_committed_count` and add a few comments or todos

File: cmd/thor/node/packer_loop.go (hunk 4 above)
Comment (libotony) [reply cmd/thor/node/packer_loop.go:123]: How about capturing it in a ticket? Seems fine to me.

Comment (otherview) [reply cmd/thor/node/packer_loop.go:123]: Yep, created https://github.com/vechain/protocol-board-repo/issues/209 👍

File: cmd/thor/node/node.go [hunk 5]
```diff
@@ -402,11 +402,14 @@ func (n *Node) processBlock(newBlock *block.Block, stats *blockStats) (bool, err
...
 		switch {
-		case err == errKnownBlock || err == errBFTRejected:
+		case err == errKnownBlock:
 			stats.UpdateIgnored(1)
 			return false, nil
 		case consensus.IsFutureBlock(err) || err == errParentMissing || err == errBlockTemporaryUnprocessable:
 			stats.UpdateQueued(1)
+		case err == errBFTRejected:
+			// TODO: capture metrics
+			log.Debug(fmt.Sprintf("block rejected by BFT engine\n%v", newBlock.Header()))
```
Comment (otherview) [inline cmd/thor/node/node.go:412]: Nice 👍

**Pull request merged**